go-controller/pkg/node/healthcheck_node.go

```diff
@@ -1,62 +1,102 @@
 package node
 
 import (
+	"context"
 	"fmt"
 	"net/http"
+	"os"
 	"sync"
 	"time"
 
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
 	"github.com/pkg/errors"
 
 	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ktypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/klog/v2"
 	"k8s.io/utils/clock"
 )
 
+var updateInterval time.Duration = 500 * time.Millisecond
+
 type proxierHealthUpdater struct {
-	address  string
-	nodeRef  *kapi.ObjectReference
-	recorder record.EventRecorder
-	c        clock.Clock
-	healthy  bool
+	address     string
+	nodeRef     *kapi.ObjectReference
+	recorder    record.EventRecorder
+	c           clock.Clock
+	healthy     bool
+	lastUpdated time.Time
+	kubeClient  kubernetes.Interface
 }
 
 // newNodeProxyHealthzServer creates and returns a new proxier health server
 // if the HealthzBindAddress configuration is set. Cloud load balancers use this
 // health check to determine if the node is available for services with ClusterIP
 // traffic policy.
-func newNodeProxyHealthzServer(nodeName, address string, eventRecorder record.EventRecorder) *proxierHealthUpdater {
+func newNodeProxyHealthzServer(nodeName, address string, eventRecorder record.EventRecorder, kubeClient kubernetes.Interface) *proxierHealthUpdater {
 	return &proxierHealthUpdater{
-		address:  address,
-		recorder: eventRecorder,
-		c:        clock.RealClock{},
-		healthy:  true,
+		address:     address,
+		recorder:    eventRecorder,
+		c:           clock.RealClock{},
+		healthy:     true,
+		lastUpdated: time.Time{},
 		nodeRef: &kapi.ObjectReference{
 			Kind:      "Node",
 			Name:      nodeName,
 			UID:       ktypes.UID(nodeName),
 			Namespace: "",
 		},
+		kubeClient: kubeClient,
+	}
+}
+
+func (phu *proxierHealthUpdater) isOvnkNodePodTerminating() bool {
+	namespace := config.Kubernetes.OVNConfigNamespace
+	podName := os.Getenv("POD_NAME")
+	if len(podName) == 0 {
+		klog.Errorf("Found empty env variable POD_NAME")
+		return false
 	}
+
+	pod, err := phu.kubeClient.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
```

Comment:
I always forget; is this coming from the informer cache, or is this a straight get to the apiserver?

Reply:
Good point. This is from the apiserver. In order to use informer cache, I should go through node watch factory instead. Let me do that.

Reply:
So if I use the informer cache, I don't have to rate limit the calls to `GetPod`, as the rate limiting happens between the informer cache and the API server, right?

Reply:
Yep, that's correct.

Reply:
In general we always want to use the informer (except in specific cases) cache because going to the apiserver is expensive and places more load on it.